import-service/importFileParser.js

```diff
@@ -0,0 +1,45 @@
+const AWS = require('aws-sdk');
+const csv = require('csv-parser');
+const { IMPORT_BUCKET_NAME } = require('./constants');
+
+module.exports = async (event) => {
+    const s3 = new AWS.S3({region: 'eu-west-1'});
+    const {Records} = event;
+    console.log('importFileParser Event', event);
+    const readStream = async (stream, Key)=> {
+        return new Promise((resolve, reject)=>{
+            try{
+
+                stream.pipe(csv())
+                    .on('data', data=>{
+                        console.log(data);
+                    })
+                    .on('end', async()=>{
+                        console.log(`Copy from ${IMPORT_BUCKET_NAME}/${Key}`);
+                        const destinationKey = Key.replace('uploaded', 'parsed');
+                        await s3.copyObject({
+                            Bucket: IMPORT_BUCKET_NAME,
+                            CopySource: `${IMPORT_BUCKET_NAME}/${Key}`,
+                            Key: destinationKey
+                        }).promise();
+                        console.log(`Copied into ${IMPORT_BUCKET_NAME}/${destinationKey}`);
+                        resolve();
+                    });
+            }
+            catch (error) {
+                reject(error)
+            }
+        });
+    }
+
+    for (const record of Records){
+        const {key: Key} = record.s3.object;
+    
+        const stream = s3.getObject({
```

Comment:
You don't actually need to get a list of all available files and try to parse and move them to another directory. The incoming event already contains the key and other metadata for the newly added entries. In my opinion lambda function should process only above mentioned records (from S3Event), because you may have invalid .csv file in your `/upload` directory and the function will try to parse it every time when new files are added, because if it gets an error while parsing the file won't move to another directory.
```
{
    "Records": [
        {
            "eventVersion": "2.1",
            "eventSource": "aws:s3",
            ...
            "s3": {
                "s3SchemaVersion": "1.0",
                "configurationId": "import-service-dev-importFileParser-9eceafef0fe575eb52543c9070542d86",
                "bucket": {
                    "name": "BUCKET_NAME",
                    ...
                },
                "object": {
                    "key": "FOLDER_NAME/FILE_NAME",
                    ...
                }
            }
        }
    ]
}
```